### PR TITLE
feat: docusaurus v2

### PR DESCRIPTION
**Pull request by eriveltonelias**
Sorry for this long PR 😅, but a lot of changes were necessary to migrate Docusaurus v1 to v2.
on this link https://v2.docusaurus.io/docs/migrating-from-v1-to-v2 we can see the instructions to migrate, may it help for review.

Screen Shot 2020-02-18 at 07 44 21
Screen Shot 2020-02-18 at 07 44 29
Screen Shot 2020-02-18 at 07 44 34

Screen Shot 2020-02-18 at 07 44 16

### CODE REVIEW

**Comment by eriveltonelias** [comment]
@satya164 yeah, you're right. Just removed this one.

**Comment by eriveltonelias** [comment]
> There seem to be 2 folders, `website/` and `website1`. Why are both folders needed?
> 
thanks! I'll remove the `website1`
> Also I would like to remove docs from `website/docs` and keep everything in versioned like in current repo because having duplicated docs in `docs/` always confuses users.

if we remove the `docs/` will "remove" the 'next' version, so can I remove this one too?

**Comment by satya164** [comment]
> if we remove the docs/ will "remove" the 'next' version, so can I remove this one too?

yea, we don't have a next right now

**Comment by eriveltonelias** [comment]
done :)

File: website/versioned_docs/version-5.x/nesting-navigators.md [hunk 1]
```diff
@@ -59,10 +58,6 @@ For example, if you have a stack inside a drawer navigator, the drawer's `openDr
 
 Similarly, if you have a tab navigator inside stack navigator, the screens in the tab navigator will get the `push` and `replace` methods for stack in their `navigation` prop.
 
-### Nested navigators don't receive parent's events
-
-For example, if you have a stack navigator nested inside a tab navigator, the screens in the stack navigator won't receive the events emitted by the parent tab navigator such as (`tabPress`) when using `navigation.addListener`. To receive events from parent navigator, you can explicitly listen to parent's events with `navigation.dangerouslyGetParent().addListener`.
```
Comment (WoLewicki) [inline website/versioned_docs/version-5.x/nesting-navigators.md]: Why delete this?

File: website/versioned_docs/version-5.x/stack-navigator.md [hunk 2]
```diff
@@ -126,21 +123,6 @@ headerStyle: {
 
 To set a custom header for all the screens in the navigator, you can specify this option in the `screenOptions` prop of the navigator.
 
-If you want your custom header to animate with screen transitions and want to keep `headerMode` as `float`, you can interpolate on the `scene.progress.current` and `scene.progress.next` props. For example, following will cross-fade the header:
```
Comment (WoLewicki) [inline website/versioned_docs/version-5.x/stack-navigator.md]: Why delete this too?

File: website/versioned_docs/version-5.x/use-linking.md [hunk 3]
```diff
@@ -43,7 +41,7 @@ export default function App() {
   }
 
   return (
-    <NavigationContainer initialState={initialState} ref={ref}>
+    <NavigationContainer initialState={initialState}>
```
Comment (WoLewicki) [inline website/versioned_docs/version-5.x/use-linking.md]: Why remove ref? It is used in `useLinking`.

**Comment by satya164** [comment]
I think these deletions/changes are because of bad merge. Let's make a pass to see the documentation is the same as master and then merge :)

**Comment by eriveltonelias** [comment]
yeah, @satya164 is right, @WoLewicki. Basically, some things were needed to change when migrating v1 to v2.
eg: 
```js
-- id: VERSION-FILE_NAME
-- original_id: FILE_NAME
```
the original_id was removed and the `id` was needed to remove the version reference and when I started working on this upgrade was on last week, so a lot of changes happened :D. 

and `docs` now should be on `/website` so when I tried to merge this today some bad merges happen, but I don't think they were deleted

**Comment by satya164** [comment]
I've fixed the merge related issues. I was running  it locally, few things:
- Seems clicking the blog link crashes the website and I see empty page
- Clicking on some links doesn't work (e.g. Continue to "Hello React Navigation" to start writing some code.) in "Getting Started". I think need to replace the ".html" with ".md"
- Hovering over the buttons in homepage has white text with light background, can we make the text purple?
- We're missing this section "React Navigation is built and funded by Expo & Software Mansion, with contributions from the community."
- The copyright text looks a bit weird due to small top margin, maybe add some margin there? or remove that text, idc
- Regarding dark theme for code snippets in light background. I've astigmatism and it feels too harsh to my eyes. Can we change it to light theme?
- The code snippets with `diff` don't have syntax highlighting, check the "upgrading from 4.x" doc for an example
- Purple accent color may not be best with dark theme. Maybe we can lighten it up a bit for the dark theme
- Since everything is inside the website folder now, maybe we can just move everything to root of the project? Is there any disadvantage? Currently I keep running `yarn` at root by mistake instead of `cd website/` first
- Not related to this PR, but can we remove "Next" from versions page? Since we don't have a "next" version right now

**Comment by eriveltonelias** [comment]
> I've fixed the merge related issues. I was running it locally, few things:
> 
thanks ❤️
> * Seems clicking the blog link crashes the website and I see empty page
> * Clicking on some links doesn't work (e.g. Continue to "Hello React Navigation" to start writing some code.) in "Getting Started". I think need to replace the ".html" with ".md"
> * Hovering over the buttons in homepage has white text with light background, can we make the text purple?
>
yes

> * We're missing this section "React Navigation is built and funded by Expo & Software Mansion, with contributions from the community."
> * The copyright text looks a bit weird due to small top margin, maybe add some margin there? or remove that text, idc
> * Regarding dark theme for code snippets in light background. I've astigmatism and it feels too harsh to my eyes. Can we change it to light theme?
>
for sure, I'll change
> * The code snippets with `diff` don't have syntax highlighting, check the "upgrading from 4.x" doc for an example
> * Purple accent color may not be best with dark theme. Maybe we can lighten it up a bit for the dark theme
> * Since everything is inside the website folder now, maybe we can just move everything to root of the project? Is there any disadvantage? Currently I keep running `yarn` at root by mistake instead of `cd website/` first
>
yeah, we can move everything to root now. 
> * Not related to this PR, but can we remove "Next" from versions page? Since we don't have a "next" version right now
> 
yeah


Thanks for all feedback @satya164.

**Comment by eriveltonelias** [comment]
- [ ] Seems clicking the blog link crashes the website and I see empty page

- [ ] Clicking on some links doesn't work (e.g. Continue to "Hello React Navigation" to start writing some code.) in "Getting Started". I think need to replace the ".html" with ".md"

- [ ] Hovering over the buttons in homepage has white text with light background, can we make the text purple?
- [ ] We're missing this section "React Navigation is built and funded by Expo & Software Mansion, with contributions from the community."
- [ ] The copyright text looks a bit weird due to small top margin, maybe add some margin there? or remove that text, idc
- [ ] Regarding dark theme for code snippets in light background. I've astigmatism and it feels too harsh to my eyes. Can we change it to light theme?
- [ ] The code snippets with `diff` don't have syntax highlighting, check the "upgrading from 4.x" doc for an example
- [ ] Purple accent color may not be best with dark theme. Maybe we can lighten it up a bit for the dark theme
- [ ] Since everything is inside the website folder now, maybe we can just move everything to root of the project? Is there any disadvantage? Currently I keep running `yarn` at root by mistake instead of `cd website/` first
- [ ] Not related to this PR, but can we remove "Next" from versions page? Since we don't have a "next" version right now

**Comment by eriveltonelias** [comment]
@satya164 
Can you check those topics again, please? 

- Seems clicking the blog link crashes the website and I see empty page

- Clicking on some links doesn't work (e.g. Continue to "Hello React Navigation" to start writing some code.) in "Getting Started". I think need to replace the ".html" with ".md" 

- Purple accent color may not be best with a dark theme. Maybe we can lighten it up a bit for the dark theme ( **Do you have a suggestion for what color in here** 😅)?

**Comment by satya164** [comment]
> Can you check those topics again, please?

Kapture 2020-02-21 at 14 41 09

> Do you have a suggestion for what color in here

#a697ce looks good contrast wise

**Comment by satya164** [comment]
Just fixed videos not playing because docusaurus now uses MDX and it needs to be `autoPlay` instead of `autoplay`

**Comment by satya164** [comment]
Turns out the blog doesn't load because the chunk name has "ad" in it which gets blocked by uBlock origin 🤦‍♂

**Comment by satya164** [comment]
It's live https://reactnavigation.org/

Thanks so much for working on this :D

**Comment by satya164** [comment]
Seems the new version breaks all the links because of removing `.html` suffix. There were some other issues which I tried to fix here https://github.com/react-navigation/react-navigation.github.io/commits/docusaurus-v2

But I didn't notice that translations don't work 😱

Until we figure that out, I've reverted to v1 and put the v2 branch here https://github.com/react-navigation/react-navigation.github.io/commits/docusaurus-v2

**Comment by eriveltonelias** [comment]
> It's live https://reactnavigation.org/
> 
> Thanks so much for working on this :D

❤️😭

**Comment by eriveltonelias** [comment]
> Seems the new version breaks all the links because of removing `.html` suffix. There were some other issues which I tried to fix here https://github.com/react-navigation/react-navigation.github.io/commits/docusaurus-v2
> 
> But I didn't notice that translations don't work 😱

😩haha

sure, let me work on that, btw our account on Crowdin was suspended, do you have an idea why?

**Pull request merged**